### PR TITLE
Initial work to start switch to core use in C++ side

### PR DESCRIPTION
**Pull request by kdt3rd**
Each commit will hopefully break down what are going to be some large changes. This represents what will be a longer chain of conversion of the C++ library to use the core library internally to gain the benefits present there.

Website preview: https://openexr--1719.org.readthedocs.build/en/1719/

### CODE REVIEW

File: src/lib/OpenEXRCore/chunk.c [hunk 1]
```diff
@@ -742,8 +742,8 @@ exr_read_scanline_chunk_info (
 
     if (!cinfo) return ctxt->standard_error (ctxt, EXR_ERR_INVALID_ARGUMENT);
 
-    if (part->storage_mode == EXR_STORAGE_TILED ||
```
Comment (meshula) [inline src/lib/OpenEXRCore/chunk.c:732]: yikes! I think this a bug fix to fastrack to main

File: src/lib/OpenEXRCore/chunk.c [hunk 2]
```diff
@@ -1002,8 +1002,8 @@ exr_read_tile_chunk_info (
 
     if (!cinfo) return ctxt->standard_error (ctxt, EXR_ERR_INVALID_ARGUMENT);
 
-    if (part->storage_mode == EXR_STORAGE_SCANLINE ||
```
Comment (meshula) [inline src/lib/OpenEXRCore/chunk.c:992]: yikes! I think this a bug fix to fastrack to main

File: src/lib/OpenEXRCore/parse_header.c [hunk 3]
```diff
@@ -1676,8 +1683,13 @@ check_populate_version (
 
     attrsz = (int32_t) one_to_native32 ((uint32_t) attrsz);
     if (attrsz != 1)
-        return ctxt->print_error (
```
Comment (meshula) [inline src/lib/OpenEXRCore/parse_header.c:1667]: possible fastrack

File: src/lib/OpenEXR/ImfInputFile.cpp [hunk 4]
```diff
@@ -144,6 +145,10 @@ InputFile::InputFile (InputPartData* part)
     initialize ();
 }
 
+InputFile::~InputFile ()
```
Comment (meshula) [inline src/lib/OpenEXR/ImfInputFile.cpp:148]: are you gonna put something in here? this might be just an =default in the header?

**Review by meshula — COMMENTED** [review]
I realize this is all ground work, but I see a reasonable amount of "yikes! bug fixed!" going on here. If you wanna grow this PR for a while, could I ask that we fasttrack the two "yikes" bugs I marked in core to main?

**Comment by kdt3rd** [comment]
yes, will talk around my plan at the TSC meeting tomorrow

File: src/lib/OpenEXR/ImfInputFile.cpp (hunk 4 above)
Comment (kdt3rd) [reply src/lib/OpenEXR/ImfInputFile.cpp:148]: oh, probably, I always get weird with virtual tables and force them to be in a specific translation unit. However, as soon as I remove the generic input file base class, will remove the vtable entirely, it is not needed / helpful (and will switch to rule of 0)

**Comment by kdt3rd** [comment]
splitting apart and moving to new branch, closing this PR

**Pull request closed without merging**